### ISSUE TITLE
Widgets screen: Provide more meaningful labels for the ARIA landmark regions

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#25864]
**Describe the bug**

The new Widgets screen uses ARIA landmark regions to split the page in meaningful sections users of assistive technology can navigate with ease. ARIA landmark regions are navigational tools and need to be labelled in a meaningful way to let users understand what the main purpose of a specific section is. The block editor uses ARIA landmark regions as well and their implementation has been recently abstracted to use a component: the "interface skeleton" component

However, to my understanding, right now the Widgets screen landmarks are labeled based on the default labels, which aren't that meaningful:

https://github.com/WordPress/gutenberg/blob/029b1d3ce7ee9b32b49396c1a030b2f22b10b0ae/packages/interface/src/components/interface-skeleton/index.js#L39-L52

They're meant as a generic fallback when no other labels have been passed. See in the screenshot below how these landmarks are currently labelled (the last three ones):

<img width="1296" alt="Screenshot 2020-10-05 at 18 12 58" src="https://user-images.githubusercontent.com/1682452/95215790-ab74bf80-07f1-11eb-92ca-5c903533e2a0.png">

- "Header": doesn't tell users what the purpose of the top bar is, it's way too generic
- "Content": it's better, but can be further improved to indicate this area is the one with the list of Widgets and Blocks
- "Settings": a bit generic but I wouldn't know how to better communicate the purpose of the sidebar

These labels should be made a bit more meaningful by passing specific labels to the layout component used for the Widgets screen, [in the same way it's done for the block editor](https://github.com/WordPress/gutenberg/blob/029b1d3ce7ee9b32b49396c1a030b2f22b10b0ae/packages/edit-post/src/components/layout/index.js#L55-L67).

**To reproduce**
Use a screen reader and check the list of landmarks or navigate through them with the specific keyboard shortcuts provided by the screen reader. For example:
- VoiceOver:
  - Control Option U to invoke the VoiceOver "rotor"
  - observe the landmark names given by their labels
  - for completeness: left or right arrows to get to the Landmark panel
  - down or up arrow to select a landmark
  - Enter or Spacebar to jump to the selected landmark
- NVDA
  - NVDA key + F7 to get the "Elements list" dialog
  - navigate the dialog with the Tab and arrow keys to get to the Landmark tab
  - observe the landmark names given by their labels
  - for completeness: choose one of the landmarks with the arrow keys and press Enter to jump to it
  - it is also possible to navigate through landmark by pressing the D key or Shift + D key

**Expected behavior**
The landmarks to have more meaningful names.

**Editor version (please complete the following information):**
- WordPress version: 5.6 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? plugin
- If the Gutenberg plugin is installed, which version is it? latest 9.1.0 master

**Comment by afercia** [comment]
Note: regarding the default labels, positional references like:

`leftSidebar: __( 'Left sidebar' )`

should be avoided because "Left sidebar" is meaningless for screen reader users. While the concept of "left" is certainly understandable, "Left sidebar" doesn't tell anything about the purpose of such a landmark region. To keep it generic (since this is a fallback) something like "Extra" or "Aside" or "Complementary" would be more appropriate.
